File: sdk/python/hitechcloud/resources/passkey_v2.py
"""PasskeyV2 resource (6 endpoints)"""

from .base import BaseResource

class PasskeyV2Resource(BaseResource):
    """PasskeyV2: passkey-based MFA management"""

    def get_mfa_status(self, user_type: str, user_id: int) -> dict:
        """GET /api/passkeyv2/status/{user_type}/{user_id} - Get MFA status"""
        return self._http.get(f"/api/passkeyv2/status/{user_type}/{user_id}")

    def list_credentials(self, user_type: str, user_id: int) -> dict:
        """GET /api/passkeyv2/credentials/{user_type}/{user_id} - List passkey credentials"""
        return self._http.get(f"/api/passkeyv2/credentials/{user_type}/{user_id}")

    def delete_credential(self, user_type: str, user_id: int, **kwargs) -> dict:
        """POST /api/passkeyv2/credentials/{user_type}/{user_id}/delete - Delete credential"""
        return self._http.post(f"/api/passkeyv2/credentials/{user_type}/{user_id}/delete", kwargs)

    def send_email_otp(self, **kwargs) -> dict:
        """POST /api/passkeyv2/email_otp/send - Send email OTP"""
        return self._http.post("/api/passkeyv2/email_otp/send", kwargs)

    def verify_email_otp(self, **kwargs) -> dict:
        """POST /api/passkeyv2/email_otp/verify - Verify email OTP"""
        return self._http.post("/api/passkeyv2/email_otp/verify", kwargs)

    def disable_mfa(self, user_type: str, user_id: int) -> dict:
        """POST /api/passkeyv2/disable/{user_type}/{user_id} - Disable MFA"""
        return self._http.post(f"/api/passkeyv2/disable/{user_type}/{user_id}")

File: sdk/ruby/lib/hitechcloud/resources/passkey_v2.rb
module HiTechCloud
  # PasskeyV2 resource
  class PasskeyV2Resource < BaseResource

    # Get status
    def status(user_type, user_id)
      @http.get("/api/passkeyv2/status/%{user_type}/%{user_id}" % { user_type: user_type, user_id: user_id })
    end

    # List credentials
    def credentials(user_type, user_id)
      @http.get("/api/passkeyv2/credentials/%{user_type}/%{user_id}" % { user_type: user_type, user_id: user_id })
    end

    # Delete credential
    def delete_credential(user_type, user_id, data)
      @http.post("/api/passkeyv2/credentials/%{user_type}/%{user_id}/delete" % { user_type: user_type, user_id: user_id }, data)
    end

    # Send email OTP
    def send_email_otp(data)
      @http.post("/api/passkeyv2/email_otp/send", data)
    end

    # Verify email OTP
    def verify_email_otp(data)
      @http.post("/api/passkeyv2/email_otp/verify", data)
    end

    # Disable MFA
    def disable_mfa(user_type, user_id)
      @http.post("/api/passkeyv2/disable/%{user_type}/%{user_id}" % { user_type: user_type, user_id: user_id })
    end

  end
end
